src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectionRuleOptionsTests.cs

```diff
@@ -829,6 +829,193 @@ public Task CollectionRuleOptions_LoadProfilerAction_ClsidPropertyValidation()
+                ex =>
+                {
+                    string[] failures = ex.Failures.ToArray();
+                    Assert.Single(failures);
+                    VerifyRequiredMessage(failures, 0, nameof(SetEnvironmentVariableOptions.Name));
+                });
+
+            await ValidateFailure(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddSetEnvironmentVariableAction(
+                        configureOptions: opts =>
+                        {
+                            opts.Name = string.Empty;
+                        });
+                },
+                ex =>
+                {
+                    string[] failures = ex.Failures.ToArray();
+                    Assert.Single(failures);
+                    VerifyRequiredMessage(failures, 0, nameof(SetEnvironmentVariableOptions.Name));
+                });
+
+            await ValidateFailure(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddSetEnvironmentVariableAction(
+                        configureOptions: opts =>
+                        {
+                            opts.Name = "    ";
+                        });
+                },
+                ex =>
+                {
+                    string[] failures = ex.Failures.ToArray();
+                    Assert.Single(failures);
+                    VerifyRequiredMessage(failures, 0, nameof(SetEnvironmentVariableOptions.Name));
+                });
+        }
+
+        [Fact]
+        public Task CollectionRuleOptions_GetEnvironmentVariable_RoundTrip()
+        {
+            const string VariableName = "MyGettingVar";
+
+            return ValidateSuccess(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddGetEnvironmentVariableAction(
+                        configureOptions: opts =>
+                        {
+                            opts.Name = VariableName;
+                        });
+                },
+                ruleOptions =>
+                {
+                    Assert.Single(ruleOptions.Actions);
+                    ruleOptions.VerifyGetEnvironmentVariableAction(0, VariableName);
+                });
+        }
+
+        [Fact]
+        public async Task CollectionRuleOptions_GetEnvironmentVariable_NamePropertyValidation()
+        {
+            await ValidateFailure(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddGetEnvironmentVariableAction(
+                        configureOptions: opts =>
```

Comment:
nit: With `configureOptions` having a `null` default, you don't need to specify this anonymous function.

Reply:
These are no longer fluent-style